src/ui/components/navigation/navigationcomponent.js

```diff
@@ -479,8 +480,13 @@ export default class NavigationComponent extends Component {
     }
 
     // We want to persist the params from the existing URL to the new
-    // URLS we create.
+    // URLS we create, except for facets/filters/pagination
     params.set('tabOrder', this._tabOrder);
-    return baseUrl + '?' + params.toString();
+    const filteredParams = filterParamsForExperienceLink(
+      params,
+      this.componentManager.getComponentNamesForComponentTypes.bind(this.componentManager)
```

Comment:
Minor nit: let's use an inline function here, that will save us the 'this' binding and is a bit more readable (I think). types => this.componentManager.getComponentNamesForComponentTypes(types). 
Same thing for the other invocations of filterParamsForExperienceLink.